frontend/pkg/frontend/frontend.go

```diff
@@ -230,7 +233,7 @@ func (f *Frontend) ArmResourceRead(writer http.ResponseWriter, request *http.Req
 	if err != nil {
 		if errors.Is(err, database.ErrNotFound) {
 			f.logger.Error(fmt.Sprintf("existing document not found for cluster: %s", resourceID))
-			writer.WriteHeader(http.StatusNoContent)
+			writer.WriteHeader(http.StatusNotFound)
```

Comment:
These need to return ARM-formatted errors/CloudErrors

Reply:
Were you going to make it a cloud error in this PR or are you just focused on status codes here?  I can pick it up if not.

Reply:
Oh sorry - I'll update it for cloud errors for this PR